src/useMultiRootEditor.tsx

```diff
@@ -0,0 +1,508 @@
+				}
+
+				newContent[ rootName ] = editor!.getData( { rootName } );
+			} );
+
+		modelDocument.differ.getChangedRoots()
+			.forEach( changedRoot => {
+				// Ignore added and removed roots. They are handled by a different function.
+				// Only register if roots attributes changed.
+				if ( changedRoot.state ) {
+					if ( newContent[ changedRoot.name ] !== undefined ) {
+						delete newContent[ changedRoot.name ];
+					}
+
+					return;
+				}
+
+				const rootName = changedRoot.name;
+
+				newAttributes[ rootName ] = editor!.getRootAttributes( rootName );
+			} );
+
+		if ( Object.keys( newContent ).length ) {
+			setContent( previousContent => ( { ...previousContent, ...newContent } ) );
+
+			shouldUpdateEditor.current = false;
+		}
+
+		if ( Object.keys( newAttributes ).length ) {
+			setAttributes( previousAttributes => ( { ...previousAttributes, ...newAttributes } ) );
+
+			shouldUpdateEditor.current = false;
+		}
+
+		/* istanbul ignore else */
+		if ( props.onChange ) {
+			props.onChange( event, editor! );
+		}
+	};
+
+	/**
+	 * Callback function for handling an added root.
+	 */
+	const onAddRoot = ( editor: MultiRootEditor, evt: EventInfo, root: RootElement ): void => {
+		const rootName = root.rootName;
+
+		const reactElement = <div
+			ref={ el => {
+				if ( el ) {
+					const editable = editor.ui.view.createEditable( rootName, el );
+
+					editor.ui.addEditable( editable );
+
+					editor.editing.view.forceRender();
+				}
+			} }
+			key={ rootName }
+			id={ rootName }
+		/>;
+
+		setContent( previousContent =>
+			( { ...previousContent, [ rootName ]: editor!.getData( { rootName } ) } )
+		);
+
+		setAttributes( previousAttributes =>
+			( { ...previousAttributes, [ rootName ]: editor!.getRootAttributes( rootName ) } )
+		);
+
+		setElements( previousElements => [ ...previousElements, reactElement ] );
+	};
+
+	/**
+	 * Callback function for handling a detached root.
+	 */
+	const onDetachRoot = ( editor: MultiRootEditor, evt: EventInfo, root: RootElement ): void => {
+		const rootName = root.rootName;
+
+		setElements( previousElements => previousElements.filter( element => element.props.id !== rootName ) );
+
+		setContent( previousContent => {
+			const { [ rootName! ]: _, ...newContent } = previousContent;
+
+			return { ...newContent };
+		} );
+
+		setAttributes( previousAttributes => {
+			const { [ rootName! ]: _, ...newAttributes } = previousAttributes;
+
+			return { ...newAttributes };
+		} );
+
+		editor!.detachEditable( root );
+	};
+
+	/**
+	 * Creates an editor from the element and configuration.
+	 *
+	 * @param initialContent The initial content.
+	 * @param config CKEditor 5 editor configuration.
+	 */
+	const _createEditor = (
+		initialContent: Record<string, string> | Record<string, HTMLElement>,
+		config: EditorConfig
+	): Promise<MultiRootEditor> => {
+		return props.editor.create( initialContent, config )
+			.then( ( editor: MultiRootEditor ) => {
+				if ( 'disabled' in props ) {
```

Comment:
Do we have support for this property in other parts of the code? Can I change it and will it change the editor state?

Reply:
TBH I copied this logic from the "basic" integration. I'm not sure whether we should add something extra here.

Reply:
I added additional logic and unit tests.